inspectit-oce-agent/src/system-test/java/rocks/inspectit/oce/instrumentation/UserInstrumentationWithMetricsTest.java

```diff
@@ -0,0 +1,71 @@
+package rocks.inspectit.oce.instrumentation;
+
+import io.opencensus.common.Scope;
+import io.opencensus.stats.AggregationData;
+import io.opencensus.tags.TagContextBuilder;
+import io.opencensus.tags.TagKey;
+import io.opencensus.tags.TagValue;
+import io.opencensus.tags.Tags;
+import org.junit.jupiter.api.Test;
+import rocks.inspectit.oce.TestUtils;
+
+import java.util.HashMap;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class UserInstrumentationWithMetricsTest extends InstrumentationSysTestBase {
+
+
+    void invocationCount() {
+
+    }
+
+    void responseTimeMeasuring() throws Exception {
+        Thread.sleep(100);
+    }
+
+    @Test
+    void invocationCounterTest() {
+        TagContextBuilder builder = Tags.getTagger().currentBuilder()
+                .put(TagKey.create("user_tag"), TagValue.create("user_value"));
+        try (Scope tcs = builder.buildScoped()) {
+            for (int i = 0; i < 7; i++) {
+                invocationCount();
+            }
+        }
+        TestUtils.waitForOpenCensusQueueToBeProcessed();
+
+        HashMap<String, String> countTags = new HashMap<>();
```

Comment:
Can you refactor this a little bit for improving readabillity - separating the assertion from the value gathering when it is quite long.

Like:
```
Map<String, String> countTags = ImmutableMap.of("user_tag", "user_value", "method_name", "invocationCount");
long invocationCount = ((AggregationData.CountData) TestUtils.getDataForView("my/invocation/count", countTags)).getCount();
assertThat(invocationCount).isEqualTo(7);

Map<String, String> sumTags = ImmutableMap.of("method_name", "invocationCount");
long invocationSum = ((AggregationData.SumDataLong) TestUtils.getDataForView("my/invocation/sum", sumTags)).getSum();
assertThat(invocationSum).isEqualTo(7 * 42);
```